code/modules/mining/mine_turfs.dm

```diff
@@ -602,6 +602,17 @@ var/global/list/rockTurfEdgeCache
 	icon_state = "asteroid_dug"
 	return
 
+/turf/simulated/floor/plating/asteroid/singularity_pull(S, current_size)
+	if(current_size == STAGE_THREE)
+		if(prob(30))
+			gets_dug()
+	else if(current_size == STAGE_FOUR)
+		if(prob(50))
+			gets_dug()
+	else if(current_size >= STAGE_FIVE)
+		if(prob(90))
+			gets_dug()

```

Comment:
It's a pull, it pulls the sand. I'll leave transforming the turf to space to singularity eating.